doc/plutus/explanations/plutus-tools-component-descriptions.rst

```diff
@@ -0,0 +1,321 @@
+| State        | Work in progress to support all the functionality from        | 
+|              | plutus-ledger-constraints.                                    |
++--------------+---------------------------------------------------------------+
+
+6. Plutus chain index
+---------------------------------
+
+`Plutus chain index <https://github.com/input-output-hk/plutus-apps/tree/main/plutus-chain-index-core>`_ 
+is an application for indexing useful data from the Cardano blockchain that is to be 
+used in the Contract Monad. The main design is for indexing information related to the 
+UTxO set in order to keep the amount of indexed information proportional to the 
+UTxO set. The component is not meant to index historical information, contrary to 
+the purpose of cardano-db-sync. 
+
++--------------+---------------------------------------------------------------+
+| Plutus chain index                                                           |
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-chain-index-core``                       |
+|              | ``plutus-apps/plutus-chain-index``                            |
++--------------+---------------------------------------------------------------+
+| Dependencies | cardano-api, plutus-ledger-api, plutus-ledger,                |
+|              | plutus-script-utils                                           |
++--------------+---------------------------------------------------------------+
+| State        | In the process of being deprecated in favor of Marconi.       | 
++--------------+---------------------------------------------------------------+
+
+7. Contract monad (aka contract API, plutus contract)
+-----------------------------------------------------------
+
+`Contract monad <https://github.com/input-output-hk/plutus-apps/tree/main/plutus-contract>`_ 
+is a library for writing Plutus contracts. It provides the Contract Monad interface 
+for writing the off-chain part of a Plutus application that is to be interpreted 
+by an emulator or a plutus application backend (PAB). 
+
++--------------+---------------------------------------------------------------+
+| Contract monad                                                               |
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-contract``                               |
++--------------+---------------------------------------------------------------+
+| Dependencies | cardano-api, plutus-ledger-api, plutus-ledger,                |
+|              | plutus-ledger-constraints, plutus-chain-index-core            |
++--------------+---------------------------------------------------------------+
+| State        | Maintained but not extended. Unstable interface.              | 
++--------------+---------------------------------------------------------------+
+
+8. Contract monad emulator
```

Comment:
Okay, now I'm confused: what _is_ this document. Is it a list of libraries? evidently not, since we've now got multiple components in the same library. So what _is_ it?

Reply:
This comes of a list that I produced for @joseph-fajen. The idea was to list logical components in the current plutus-apps repository, which do not necessarily have a 1-1 relationship which the cabal projects. For example, `plutus-contract` contains both the API to write plutus applications, as well as the emulator to test plutus-applications. These two logical components could easily be split into two different cabal projects, hence why I listed them as separate _logical_ components.

Seems like that was maybe not a good idea?

If the alternative is to list _actual_ cabal projects, then the idea of adding this information in the README directly and importing it directly in the RTD seems like a good idea. I agree we don't want to duplicate the list of cabal projects where a user can simply look at the repository.

Reply:
@koslambrou That's helpful perspective. I think it may make sense to approach this document in stages. Maybe we can address the main comments, make some clarifications and reorganizing for stage 1. Then for stage 2, we create README files where appropriate and approach it that way.

Reply:
I've updated the page to make it clear that we're talking about logical components, and that a single Haskell package may contain multiple logical components.